server/src/main/java/org/apache/cloudstack/acl/RoleManagerImpl.java

```diff
@@ -159,9 +242,6 @@ public Role updateRole(final Role role, final String name, final RoleType roleTy
             roleVO.setName(name);
         }
         if (roleType != null) {
-            if (role.getId() <= RoleType.User.getId()) {
```

Comment:
Here was a check which determined that the role was built-in, i.e. the first 4 roles treated as built-in and no changes to these 4 roles were allowed by using enum idx without requiring a db column for that.

Reply:
@rhtyd agree that the role ids are compared against the role type enum earlier. now these roles, with ids (1,2,3,4) are marked as default roles, which is considered while updating or deleting the role. The `role` is already populated with db details.